### PR TITLE
NAS-120115 / 23.10 / Add url string formatter

### PR DESCRIPTION
**Pull request by denysbutenko**
Details in [the ticket](https://ixsystems.atlassian.net/browse/NAS-120115). For testing, go to `Credentials -> Backup Credentials` and check the `TrueNAS URL` field.

### CODE REVIEW

**Review by undsoft — CHANGES_REQUESTED** [review]
Formatters format value from formControl to present it to the user in different form. What you need is a parser to alter value in formControl itself. Currently, while user sees `http://`, actual value submitted doesn't have it.

Let's also check other places where `url` field is used.

File: src/app/modules/ix-forms/services/ix-formatter.service.ts [hunk 1]
```diff
@@ -193,4 +197,16 @@ export class IxFormatterService {
     }
     return (1024 ** (this.iecUnits.indexOf(unitStr)));
   };
+
+  stringAsUrlFormatting = (value: string): string => {
```
Comment (undsoft) [inline src/app/modules/ix-forms/services/ix-formatter.service.ts:201]: This doesn't look correct. Just having a parser should be enough.
I think there is a mistake in `ix-input`, where it doesn't update visible value after it has been parsed.
Please update `ix-input` instead and add a test case to it.

**Review by undsoft — CHANGES_REQUESTED** [review]

File: src/app/modules/ix-forms/services/ix-formatter.service.ts (hunk 1 above)
Comment (denysbutenko) [reply src/app/modules/ix-forms/services/ix-formatter.service.ts]: It makes sense. Thanks for bringing this to my attention 👍🏻

**Review by undsoft — CHANGES_REQUESTED** [review]
Whatever parser is doing probably should only happen on blur.
Right now memory inputs are broken (see Bandwidth fields in FTP service settings).

**Comment by denysbutenko** [comment]
> Whatever parser is doing probably should only happen on blur. Right now memory inputs are broken (see Bandwidth fields in FTP service settings).

Good catch. Fixed.

**Review by undsoft — APPROVED** [review]
Looks good 👍🏻

**Comment by denysbutenko** [comment]
I stuck with adding this case in unit tests. Reverting the added check.

**Pull request merged**